thunder/executors/cudnnex.py

```diff
@@ -207,6 +208,11 @@ def compute_NHWC_strides(shape):
 # And when registering for sdpa, cudnn assumes NHWC layout. (See _transform_sdpa_inputs())
 def _sdpa_enforce_input_tensor_contiguity(a: torch.Tensor) -> torch.Tensor:
     if a.stride(-1) == 1:
+        # TODO(vedaanta-nvidia): there's an inconsistency between
```

Comment:
nice catch. I think `_transform_sdpa_inputs` should not overwrite strides if already set.

So when cudnn checker function is called with TensorProxy:
- assume contiguous stride
- create cudnn execution plan (just fyi, this was removed for the time being in your previous PR)
- store in cache for potential performance boost in execution phase

And then at execution phase with actual torch tensors:
- Use tensor strides to redo cudnn graph/execution plan ([PPOPOSED] DONT OVERWRITE/ASSUME STRIDES)
- If tensor strides are already contiguous, execution benefits from cache hit
- Execute cudnn plan

I think above approach should be safe to do. What do you think?

Reply:
> Use tensor strides to redo cudnn graph/execution plan

One potential problem is that the above step can fail even though the checker was happy. Still, with your proposal, cudnnex can at least throw a very clear runtime error when that happens, which is a strict improvement.

That being said, I think the fundamental problem here is: at this moment, a Thunder executor's checkers have no access to strides, but cudnnex needs strides to accurately check an sdpa op can be supported. Am I right? If so, we should probably discuss with Thunder folks on a better, maybe longer-term fix.